### PR TITLE
Reduce memory consumption in JoinVerticesProcess::ProcessMesh() signi…

### PR DESCRIPTION
**Pull request by ockeymm**
When execute JoinVerticesProcess::ProcessMesh()， 'uniqueVertices' and 'uniqueAnimatedVertices' Array will consumption huge memory, some device will oom when execute this funcation.
For example： 
         The size of each array's element - Vertex structure is 272 bytes. If there is a mesh of 100000 Vertex and 52 animationMeshes, these two arrays will consume 100000 * 272 * 53=1441600000 memory during function operation.
         I changed the elements of the array from the Vertex structure to the index of the vertex in the original data, and then assigned the vertex to the new target by searching for the original data in the updateXMeshVertices function. In the above example, using this method, the memory usage can be reduced to 100000 * 4 * 53=21200000

### CODE REVIEW

File: code/PostProcessing/JoinVerticesProcess.cpp [hunk 1]
```diff
@@ -157,52 +157,52 @@ void updateXMeshVertices(XMesh *pMesh, std::vector<Vertex> &uniqueVertices) {
 
     // Position, if present (check made for aiAnimMesh)
     if (pMesh->mVertices) {
-        delete [] pMesh->mVertices;
+        aiVector3D *oldVertices = pMesh->mVertices;
```
Comment (turol) [inline code/PostProcessing/JoinVerticesProcess.cpp:160]: You should use `unique_ptr` instead of manual delete.

**Comment by kimkulling** [comment]
A good idea. Let's see what the CI says to it.

**Comment by kimkulling** [comment]
This will decrease the memory consumption for the process by 68. Whowow!

**Review by kimkulling — APPROVED** [review]
Looks fine for me!

**Comment by kimkulling** [comment]
Merged, thanks a lot for your contribution.

**Pull request merged**